### PR TITLE
feat(cli): new version notifications

### LINKED ISSUE #452: CLI: new version notifications
### Description of the feature or enhancement:

Print a notification if there is a new version of the CDK8s CLI.
* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by eladb**
The CLI will print a notification if there is a new version of the CDK8s CLI available in npm.

The result is cached under ~/.cdk8s-cli.version with a TTL of 30min.

Resolves #452

---

Screen Shot 2020-12-15 at 23 06 38

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/cdk8s-cli/src/cli/index.ts [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
-  .argv;
-
-if (args._.length === 0) {
-  yargs.showHelp();
+async function main() {
+  const versions = upgradeAvailable();
+  if (versions) {
+    console.error('------------------------------------------------------------------------------------------------');
+    console.error(colors.yellow(`A new version ${versions.latest} of cdk8s-cli is available (current ${versions.current}).`));
+    console.error(colors.yellow('Run "npm install -g cdk8s-cli" to install the latest version on your system.'));
```
Comment (campionfellin) [inline packages/cdk8s-cli/src/cli/index.ts:10]: Would it be useful here to also give the `brew` and `yarn` upgrade commands?

Comment (eladb) [reply packages/cdk8s-cli/src/cli/index.ts:10]: Maybe we just link to the docs?

Comment (iliapolo) [reply packages/cdk8s-cli/src/cli/index.ts:10]: Yes

File: packages/cdk8s-cli/src/cli/index.ts [hunk 2]
```diff
@@ -1,15 +1,32 @@
...
+    console.error('------------------------------------------------------------------------------------------------');
```
Comment (eladb) [inline packages/cdk8s-cli/src/cli/index.ts:11]:
```suggestion
    console.error(colors.yellow('Run "npm install -g cdk8s-cli" to install the latest version on your system.'));
    console.error(colors.yellow('For additional installation methods, see https://cdk8s.io/docs/latest/getting-started'));
    console.error('------------------------------------------------------------------------------------------------');
```

File: packages/cdk8s-cli/src/cli/index.ts (hunk 1 above)
Comment (eladb) [reply packages/cdk8s-cli/src/cli/index.ts:10]: Done

File: packages/cdk8s-cli/src/upgrades.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    } catch (e) {
+      return undefined;
+    }
+
+    // cannot determine version, return undefined.
+    if (!latest) {
+      return undefined;
+    }
+
+    writeCache(latest);
```
Comment (iliapolo) [inline packages/cdk8s-cli/src/upgrades.ts:58]: Protect this with `try catch` as well.

**Review by iliapolo — CHANGES_REQUESTED** [review]

File: packages/cdk8s-cli/src/upgrades.ts (hunk 3 above)
Comment (eladb) [reply packages/cdk8s-cli/src/upgrades.ts:58]: Good call

Comment (eladb) [reply packages/cdk8s-cli/src/upgrades.ts:58]: Addressed.

**Review by iliapolo — APPROVED** [review]

**Pull request merged**